### PR TITLE
Improve integration with Metrics plugin

### PR DESCRIPTION
**Pull request by jamesrobson-secondmind**
This improves the metrics generated:
* Support the fallback causes logic, so only causes that will be reported for a build are counted
* Adds a count of builds that failed without matching any causes, like the old visualisations did

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by PayBas** [comment]
Nice 👍
Looking forward to this in the next release.

**Comment by jamesrobson-secondmind** [comment]
So I'm not sure what to do about testing this. There's currently no tests for the `scan` function or the fallback categories and this change is very simple, so I'm going to say no tests are needed unless someone objects to that.

**Comment by PayBas** [comment]
Well I'm currently using this PR in combination with my own https://github.com/jenkinsci/build-failure-analyzer-plugin/pull/149 and while I am seeing:
```
# HELP jenkins_bfa_cause_no_matching_cause Generated from Dropwizard metric import (metric=jenkins_bfa.cause.no matching cause, type=com.codahale.metrics.Counter)
# TYPE jenkins_bfa_cause_no_matching_cause gauge
jenkins_bfa_cause_no_matching_cause 23.0
```
I'm not seeing the `jenkins_bfa_category_###` fallback categories. So either I integrated the 2 PRs incorrectly, or it's not working as intended.
Adding an automated test might help clear up this question ;)

**Comment by jamesrobson-secondmind** [comment]
This has some tests now, can it be merged?

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java [hunk 1]
```diff
@@ -219,6 +217,17 @@ public static void scan(Run build, PrintStream scanLog) {
...
+            if (!foundCauseList.isEmpty()) {
+                for (FoundFailureCause cause : foundCauseList) {
+                    logger.log(Level.SEVERE, cause.getName());
+                    incCounters(cause);
+                }
+            } else {
+                incCounters(UNKNOWNCAUSE);
+            }
+
+
```
Comment (PayBas) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/BuildFailureScanner.java:230]:
```
            if (!foundCauseList.isEmpty()) {
                incCounters(foundCauseList, PluginImpl.getInstance().isMetricSquashingEnabled());
            } else {
                incCounters(new ArrayList<>(Collections.singletonList(UNKNOWNCAUSE)),
                        PluginImpl.getInstance().isMetricSquashingEnabled());
            }
```

**Comment by PayBas** [comment]
Using this PR live for some time now. Looks good to me. But I'm not reviewer ;)

**Review by rsandell — APPROVED** [review]

**Pull request merged**